### DISCUSSION TITLE
Is there a way to speed up the startup time ?

### DISCUSSION THREAD

**Discussion by StitiFatah** [xonsh/xonsh#5151] [Q&A] [answered]
Hi, 

I'm just trying out xonsh and considering using it at my default shell, xonsh however takes approximately the same amount of time as ipython to start, it's in the 100s of ms range `time xonsh -c exit` returns `real 0m0.234s`, it's definitely not a deal breaker but not ideal when you spawn terminal windows pretty often, zsh/bash being pretty much instantaneous. 

Any way to speed this up ?

Edit : If that matters I'm on Arch + using kitty + xonsh 0.13.4 + python 3.11.3

**Comment by anki-code** [comment, marked as answer]
Take a look here - https://github.com/xonsh/xonsh/issues/3678#issuecomment-1576815718
